pkg/diagnostics/component_monitoring.go

```diff
@@ -381,6 +404,32 @@ func (c *componentMetrics) CryptoInvoked(ctx context.Context, component, operati
 	}
 }
 
+// JobExecutedSuccess records the metrics for a job execution success event.
+func (c *componentMetrics) JobExecutedSuccess(ctx context.Context, component, operation string, success bool, elapsed float64) {
```

Comment:
I think having `component` as the var name is a bit confusing bc its the `jobName` being passed in and technically the Jobs API doesn't have/use a `dapr component`. 

We may not even want the job name here for cardinality reasons cc: @JoshVanL for thoughts? Do you agree or do you think we want the job name here?